x-pack/solutions/observability/plugins/apm/public/components/app/transaction_details/distribution/index.tsx

```diff
@@ -106,6 +107,71 @@ export function TransactionDistribution({
     [history]
   );
 
+  // Parse logs state from URL params
+  const { logsColumns, logsSort, logsGrid, logsRowHeight, logsRowsPerPage, logsDensity } =
+    urlParams;
+
+  const logsTableConfig = useMemo(
+    () => ({
+      columns: logsColumns ? JSON.parse(logsColumns) : undefined,
+      sort: logsSort ? JSON.parse(logsSort) : undefined,
+      grid: logsGrid ? JSON.parse(logsGrid) : undefined,
+      rowHeight: logsRowHeight,
+      rowsPerPage: logsRowsPerPage,
+      density: logsDensity as SavedSearchTableConfig['density'],
+    }),
+    [logsColumns, logsSort, logsGrid, logsRowHeight, logsRowsPerPage, logsDensity]
+  );
+
+  // Handle logs table config changes and sync to URL
+  const onLogsTableConfigChange = useCallback(
+    (config: SavedSearchTableConfig) => {
+      const currentQuery = toQuery(history.location.search);
+
+      // Clean up sort configuration to only include columns that exist
+      const cleanedSort = config.sort?.filter((sortEntry) => {
+        const [fieldName] = sortEntry;
+        // Keep sorts for fields that are in the current columns array
+        // Also keep @timestamp as it's not removable
+        return fieldName === '@timestamp' || (config.columns && config.columns.includes(fieldName));
```

Comment:
`common/es_fields/apm`

```suggestion
        return fieldName === AT_TIMESTAMP || (config.columns && config.columns.includes(fieldName));
```